[CI] DfsProfilerIT testProfileDfs failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #5749 / windows-2019_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/dgt5ha2rxvss6)
- [elasticsearch-intake #15452 / part1](https://gradle-enterprise.elastic.co/s/zjej3z34qrd2s)
- [elasticsearch-periodic #5760 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/pigw6hhlxpsz4)
- [elasticsearch-periodic-platform-support #5749 / oraclelinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/zualgifc3mkw6)
- [elasticsearch-periodic-platform-support #5749 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/t5fggr442ftz4)
- [elasticsearch-periodic-platform-support #5749 / ubuntu-2004_platform-support-unix](https://gradle-enterprise.elastic.co/s/m4wsyzlbvzwmy)
- [elasticsearch-periodic-platform-support #5749 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/fmxcg3tu75kve)
- [elasticsearch-periodic #5760 / openjdk21_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/jvc2z4qvjwhog)
- [elasticsearch-periodic-platform-support #5749 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/7zykeijyqlix4)
- [elasticsearch-periodic #5760 / openjdk17_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/cifoc5iedg5w4)

**Reproduction Line:**
```
gradlew ":server:internalClusterTest" --tests "org.elasticsearch.search.profile.dfs.DfsProfilerIT.testProfileDfs" -Dtests.seed=7E03FF959CA490B6 -Dtests.locale=wae-Latn-CH -Dtests.timezone=America/Dominica -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.search.profile.dfs.DfsProfilerIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testProfileDfs),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.Exception: Test abandoned because suite timeout was reached.
```

**Issue Reasons:**
- [8.x] 2 consecutive failures in step windows-2019_checkpart1_platform-support-windows
- [8.x] 2 consecutive failures in step oraclelinux-9_platform-support-unix
- [8.x] 2 consecutive failures in step oraclelinux-8_platform-support-unix
- [8.x] 2 consecutive failures in step single-processor-node-tests
- [8.x] 2 consecutive failures in step rhel-8_platform-support-unix
- [8.x] 50 failures in test testProfileDfs (29.2% fail rate in 171 executions)
- [8.x] 2 failures in step windows-2019_checkpart1_platform-support-windows (100.0% fail rate in 2 executions)
- [8.x] 10 failures in step part1 (34.5% fail rate in 29 executions)
- [8.x] 2 failures in step oraclelinux-9_platform-support-unix (50.0% fail rate in 4 executions)
- [8.x] 2 failures in step ubuntu-2004_platform-support-unix (50.0% fail rate in 4 executions)
- [8.x] 2 failures in step oraclelinux-8_platform-support-unix (50.0% fail rate in 4 executions)
- [8.x] 2 failures in step openjdk23_checkpart1_java-matrix (66.7% fail rate in 3 executions)
- [8.x] 2 failures in step openjdk17_checkpart1_java-fips-matrix (66.7% fail rate in 3 executions)
- [8.x] 2 failures in step rocky-9_platform-support-unix (50.0% fail rate in 4 executions)
- [8.x] 7 failures in step part-1 (25.0% fail rate in 28 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

The failure here is interesting: 

```
  2> REPRODUCE WITH: ./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.search.profile.dfs.DfsProfilerIT" -Dtests.seed=BBC6EB782B8E4C50 -Dtests.locale=en -Dtests.timezone=Etc/UTC -Druntime.java=23
  2> java.lang.AssertionError: index=11, numBits=9
        at __randomizedtesting.SeedInfo.seed([BBC6EB782B8E4C50]:0)
        at org.apache.lucene.util.FixedBits.get(FixedBits.java:32)
        at org.apache.lucene.search.Weight$DefaultBulkScorer.scoreAll(Weight.java:347)
        at org.apache.lucene.search.Weight$DefaultBulkScorer.score(Weight.java:261)
        at org.elasticsearch.search.internal.ContextIndexSearcher.searchLeaf(ContextIndexSearcher.java:461)
        at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:788)
        at org.elasticsearch.search.internal.ContextIndexSearcher.search(ContextIndexSearcher.java:388)
        at org.elasticsearch.search.internal.ContextIndexSearcher.lambda$search$3(ContextIndexSearcher.java:368)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at org.apache.lucene.search.TaskExecutor$TaskGroup$1.run(TaskExecutor.java:120)
        at org.apache.lucene.search.TaskExecutor$TaskGroup.invokeAll(TaskExecutor.java:176)
        at org.apache.lucene.search.TaskExecutor.invokeAll(TaskExecutor.java:84)
        at org.elasticsearch.search.internal.ContextIndexSearcher.search(ContextIndexSearcher.java:372)
        at org.elasticsearch.search.internal.ContextIndexSearcher.search(ContextIndexSearcher.java:339)
        at org.elasticsearch.search.dfs.DfsPhase.singleKnnSearch(DfsPhase.java:225)
        at org.elasticsearch.search.dfs.DfsPhase.executeKnnVectorQuery(DfsPhase.java:202)
        at org.elasticsearch.search.dfs.DfsPhase.execute(DfsPhase.java:61)
        at org.elasticsearch.search.SearchService.executeDfsPhase(SearchService.java:562)
        at org.elasticsearch.search.SearchService.lambda$executeDfsPhase$2(SearchService.java:551)
        at org.elasticsearch.action.ActionRunnable$3.accept(ActionRunnable.java:79)
        at org.elasticsearch.action.ActionRunnable$3.accept(ActionRunnable.java:76)
        at org.elasticsearch.action.ActionRunnable$4.doRun(ActionRunnable.java:101)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
        at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:34)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1044)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1575)
```

could this be a concurrency bug of sorts? Looking into it today.

--- a/docs/changelog/122653.yaml
+++ b/docs/changelog/122653.yaml
@@ -1,0 +1,6 @@
+pr: 122653
+summary: Knn vector rescoring to sort score docs
+area: Vector Search
+type: bug
+issues:
+ - 119711

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -153,9 +153,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/117740
 - class: org.elasticsearch.xpack.security.authc.ldap.MultiGroupMappingIT
   issue: https://github.com/elastic/elasticsearch/issues/119599
-- class: org.elasticsearch.search.profile.dfs.DfsProfilerIT
-  method: testProfileDfs
-  issue: https://github.com/elastic/elasticsearch/issues/119711
 - class: org.elasticsearch.multi_cluster.MultiClusterYamlTestSuiteIT
   issue: https://github.com/elastic/elasticsearch/issues/119983
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT

--- a/server/src/main/java/org/elasticsearch/search/vectors/RescoreKnnVectorQuery.java
+++ b/server/src/main/java/org/elasticsearch/search/vectors/RescoreKnnVectorQuery.java
@@ -23,6 +23,7 @@ import org.elasticsearch.search.profile.query.QueryProfiler;
 
 import java.io.IOException;
 import java.util.Arrays;
+import java.util.Comparator;
 import java.util.Objects;
 
 /**
@@ -60,6 +61,7 @@ public class RescoreKnnVectorQuery extends Query implements QueryProfilerProvide
         TopDocs topDocs = searcher.search(query, k);
         vectorOperations = topDocs.totalHits.value();
         ScoreDoc[] scoreDocs = topDocs.scoreDocs;
+        Arrays.sort(scoreDocs, Comparator.comparingInt(scoreDoc -> scoreDoc.doc));
         int[] docIds = new int[scoreDocs.length];
         float[] scores = new float[scoreDocs.length];
         for (int i = 0; i < scoreDocs.length; i++) {

